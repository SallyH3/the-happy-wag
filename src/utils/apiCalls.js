export const getAnimals = async (url, options) => {
  const response = await fetch(url, options) 
  const result = await response.json()
  if(!response.ok) {
    throw Error('Error fetching data')
  }
  return result
}

//tests:
//fetch correct param
//good response
//one for response error
