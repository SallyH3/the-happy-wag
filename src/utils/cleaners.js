export const cleanContactInfo = (contactInfo) => {
  return {
    city: contactInfo.address.city,
    state: contactInfo.address.state,
    email: contactInfo.email
  }
}