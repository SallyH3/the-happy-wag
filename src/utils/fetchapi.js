// import App from '../containers/App';

// export const getAnimals = (url, options) => {
//   return fetch(url)
//   .then(response => response.json())
//   .catch(error => console.log(Error,( 'Error loading animals')))
// }