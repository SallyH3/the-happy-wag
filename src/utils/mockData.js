export const mockAnimals = [
  {
    age: "Baby",
    contact: {email: "lastpawtotherescue@yahoo.com"},
    description: "Sneak Peak!! Hey, you all!! Ducky here. I will be arriving to Wisconsin on Sunday from Tx. I am a...",
    gender: "Female",
    id: 44853384,
    name: "Ducky",
    photos: (3) [{}, {}, {}],
    species: "Dog"
  },
  {
    age: "Baby",
    contact: {email: "furrfoundation@gmail.com"},
    description: "Marcel is such a lovable little boy. He loves to have his...",
    gender: "Male",
    id: 44853380,
    name: "Marcel",
    photos: [{}],
    species: "Cat"
  }
]

export const mockSelectedAnimal =  {
  age: "Baby",
  contact: {
    email: "Animal.Services@cityofdenton.com", 
    phone: "(940) 349-7594", 
    address: {
      city: "Denton",
      state: "TX"
    },
    description: "Marcel is such a lovable little boy. He loves to have his...",
    gender: "Male",
    id: 44853380,
    name: "Marcel",
    photos: [{
      medium: "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/44868950/1/?bust=1559597198&width=300"
    }],
    species: "Cat"
  }
}

export const mockPhotos = [
  {
    medium: "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/44868950/1/?bust=1559597198&width=300"
  }
]

export const mockContactInfo = {
  email: "Animal.Services@cityofdenton.com", 
  phone: "(940) 349-7594", 
  address: {
    address1: "3717 N Elm Street",
    address2: null,
    city: "Denton",
    country: "US",
    postcode: "76205",
    state: "TX"
  }
}