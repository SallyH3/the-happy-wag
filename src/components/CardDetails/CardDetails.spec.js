import React from 'react';
import { CardDetails } from './index';
import { shallow } from 'enzyme';
import { cleanContactInfo } from '../../utils/cleaners.js';
import { createSerializer } from 'enzyme-to-json';
import { mockSelectedAnimal, mockContactInfo, mockPhotos } from '../../utils/mockData';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('CardDetails', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardDetails 
        contact={mockContactInfo}
        animals={mockSelectedAnimal}
        photos={mockPhotos}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})