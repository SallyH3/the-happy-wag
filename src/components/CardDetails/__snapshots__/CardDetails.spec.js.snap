// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should match snapshot 1`] = `
<section
  className="modal"
>
  <article
    className="main-details"
  >
    <img
      className="details-photo"
      src="https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/44868950/1/?bust=1559597198&width=300"
    />
    <h1
      className="details-name"
    />
    <p
      className="status"
    >
      Status: 
    </p>
    <p
      className="org_id"
    >
      Organization id: 
    </p>
    <p
      className="species-descr"
    />
    <p
      className="location"
    >
      Location: 
      Denton, TX 
    </p>
    <p
      className="email"
    >
      Email: 
      Animal.Services@cityofdenton.com
    </p>
  </article>
  <Link
    className="go-back"
    to="/"
  >
    â—€ Go Back
  </Link>
</section>
`;
